__tests__/app.test.js

```diff
@@ -136,34 +136,85 @@ describe('GET /api/articles/:articleID', ()=>{
                 expect(comments).toBeSortedBy('created_at', {descending: true})
             })
         })
-        test("should return error for valid but non-existent article id", ()=>{
+        test("404 should return error for valid but non-existent article id", ()=>{
             return request(app)
             .get('/api/articles/1000/comments')
             .expect(404)
             .then(({body})=>{
-                expect(body.msg).toBe("Not found")
+                expect(body.msg).toBe("Resource not found")
             })
         })
-        test("should return error for invalid article id", ()=>{
+        test("400 should return error for invalid article id", ()=>{
             return request(app)
-            .get('/api/articles/notanid')
+            .get('/api/articles/notanid/comments')
             .expect(400)
             .then(({body})=>{
                 expect(body.msg).toBe('Bad request')
             })
+        })
+        test("200 should return 200 empty array for valid article id with no comments", ()=>{
+            return request(app)
+                .get('/api/articles/2/comments')
+                .expect(200)
+                .then(({body})=>{
+                    const {comments} = body
+                    expect(comments.length).toBe(0)
+                })
         })   
-        
+    })
+
+    describe('POST /api/articles/:article_id/comments', ()=>{
+        test('should return added comment', ()=>{
+            const input = {username: "rogersop", body: "Ha-ha"}
+            return request(app)
+                .post('/api/articles/2/comments')
+                .send(input)
+                .expect(201)
+                .then(({body})=>{
+                    const {comment} = body
+                    expect(comment).toMatchObject({
+                        comment_id: expect.any(Number),
+                        body: expect.any(String),
+                        author: expect.any(String),
+                        created_at: expect.any(String)
+                        })
+                    expect(comment.article_id).toBe(2)
+                    expect(comment.votes).toBe(0)
+                    })
+            })
+        test("404 should return error for valid but non-existent article id", ()=>{
+            const input = {username: "rogersop", body: "Ha-ha"}
+            return request(app)
+            .post('/api/articles/1000/comments')
+            .send(input)
+            .expect(404)
+            .then(({body})=>{
+                expect(body.msg).toBe("Not found")
+            })
+        })
+        test("400 should return error for invalid article id", ()=>{
+            const input = {username: "rogersop", body: "Ha-ha"}
+            return request(app)
+            .post('/api/articles/notanid/comments')
+            .send(input)
+            .expect(400)
+            .then(({body})=>{
+                expect(body.msg).toBe('Bad request')
+            })
+        })
+
```

Comment:
I would also like to see the following tests also added:
Status 400, missing required field(s), e.g. no username or body properties
Status 404, username does not exist
Status 201, ignores unnecessary properties